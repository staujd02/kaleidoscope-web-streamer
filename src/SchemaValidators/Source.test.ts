import React from 'react';
import SourceValidation from './Source';

describe('The Source Validator', () => {

    let sourceState: Source;

    beforeEach(() => {
        sourceState = {
                title: "A title",
                duration: 5000,
                isEnabled: false,
                source: 'www.somewhere.com',
                sortOrder: 3
            }
    });

    it('should validate a correct source', () => {
        expect(SourceValidation.validate(sourceState)).toEqual(true);
    });

    describe('when given an invalid source', () => {
        
        describe('given the sort order is negative', () => {
            beforeEach(() => {
                sourceState.sortOrder = -110;
            });

            it('rejects it', () => {
                expect(SourceValidation.validate(sourceState)).toEqual(false);
            });
        });
        
        describe('given the duration is negative', () => {
            beforeEach(() => {
                sourceState.duration = -110;
            });

            it('rejects it', () => {
                expect(SourceValidation.validate(sourceState)).toEqual(false);
            });
        });

        describe('given the duration is zero', () => {
            beforeEach(() => {
                sourceState.duration = 0;
            });

            it('rejects it', () => {
                expect(SourceValidation.validate(sourceState)).toEqual(false);
            });
        });

        describe('given that the source is missing a stream title', () => {

            beforeEach(() => {
                (sourceState.title as any) = undefined;
            });

            it('rejects it', () => {
                expect(SourceValidation.validate(sourceState)).toEqual(false);
            });
        });

        describe('given that the source is missing a stream duration', () => {
            
            beforeEach(() => {
                (sourceState.duration as any) = undefined;
            });

            it('rejects it', () => {
                expect(SourceValidation.validate(sourceState)).toEqual(false);
            });
        });
    });
});