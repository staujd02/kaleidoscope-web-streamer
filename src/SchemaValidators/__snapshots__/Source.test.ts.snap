// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Source Validator when given an invalid source given that the source is missing a stream duration rejects it 1`] = `"Missing Property Exception: 'duration' is not defined"`;

exports[`The Source Validator when given an invalid source given that the source is missing a stream title rejects it 1`] = `"Missing Property Exception: 'title' is not defined"`;

exports[`The Source Validator when given an invalid source given the duration is negative rejects it 1`] = `"Duration Out of Bounds: -110 is not a valid duration"`;

exports[`The Source Validator when given an invalid source given the duration is zero rejects it 1`] = `"Duration Out of Bounds: 0 is not a valid duration"`;

exports[`The Source Validator when given an invalid source given the sort order is negative rejects it 1`] = `"Invalid Sort Order Exception: -110 is not a valid sort order"`;
